src/ansys/fluent/core/services/field_data.py

```diff
@@ -799,16 +808,59 @@ def get_pathlines_field_data(
             Dictionary containing a map of surface IDs to the pathline data.
             For example, pathlines connectivity, vertices, and field.
         """
-        if transaction_specs:
-            field_name = transaction_specs.specs.get("field_name")
-            surfaces = transaction_specs.specs.get("surfaces")
         if zones is None:
             zones = []
         pathlines_data = self.data[(("type", "pathlines-field"), ("field", field_name))]
         return self._returned_data._pathlines_data(
             field_name, surfaces, self.get_surface_ids(surfaces), pathlines_data
         )
 
+    def get_field_data(
+        self,
+        obj: [
+            SurfaceFieldDataRequest
+            | ScalarFieldDataRequest
+            | VectorFieldDataRequest
+            | PathlinesFieldDataRequest
+        ],
+    ):
+        """Get the surface, scalar, vector or path-lines field data on a surface."""
+        if isinstance(obj, SurfaceFieldDataRequest):
+            return self.get_surface_data(
+                data_types=obj.data_types,
+                surfaces=obj.surfaces,
+                overset_mesh=obj.overset_mesh,
+            )
+        elif isinstance(obj, ScalarFieldDataRequest):
+            return self.get_scalar_field_data(
+                field_name=obj.field_name,
+                surfaces=obj.surfaces,
+                node_value=obj.node_value,
+                boundary_value=obj.boundary_value,
+            )
+        elif isinstance(obj, VectorFieldDataRequest):
+            return self.get_vector_field_data(
+                field_name=obj.field_name,
+                surfaces=obj.surfaces,
+            )
+        elif isinstance(obj, PathlinesFieldDataRequest):
+            return self.get_pathlines_field_data(
+                field_name=obj.field_name,
+                surfaces=obj.surfaces,
+                additional_field_name=obj.additional_field_name,
+                provide_particle_time_field=obj.provide_particle_time_field,
+                node_value=obj.node_value,
+                steps=obj.steps,
+                step_size=obj.step_size,
+                skip=obj.skip,
+                reverse=obj.reverse,
+                accuracy_control_on=obj.accuracy_control_on,
+                tolerance=obj.tolerance,
+                coarsen=obj.coarsen,
+                velocity_domain=obj.velocity_domain,
+                zones=obj.zones,
```

Comment:
Dataclasses have a method to return the internal state as a dictionary. Maybe by using frozen dataclass instead of named tuple, we can simplify the above argument assignment.

Reply:
OK, namedtuple also has a method to return the internal state as dictionary.